clang/lib/Serialization/ASTReader.cpp

```diff
@@ -1444,6 +1444,74 @@ llvm::Error ASTReader::ReadSourceManagerBlock(ModuleFile &F) {
   }
 }
 
+llvm::Expected<SourceLocation::UIntTy>
+ASTReader::readSLocOffset(ModuleFile *F, unsigned Index) {
+  BitstreamCursor &Cursor = F->SLocEntryCursor;
+  SavedStreamPosition SavedPosition(Cursor);
+  if (llvm::Error Err = Cursor.JumpToBit(F->SLocEntryOffsetsBase +
+                                         F->SLocEntryOffsets[Index]))
+    return Err;
```

Comment:
I seem to recall that the Error & -> Expected conversion trips a compiler bug in some of our supported compilers. I think you need `std::move(Err)` when returning an l-value.

Reply:
Thanks for the heads-up, I'll keep that in mind if bots start failing. I think we recently bumped compiler requirements, so we might get away with this.